File: api/src/main/java/hirez/api/SessionStorage.java
package hirez.api;

import hirez.api.object.CreateSession;
import hirez.api.sessions.CachedSessionStorage;

import java.util.function.Consumer;

public interface SessionStorage {
    SessionStorage DEFAULT = CachedSessionStorage.create();

    String get() throws NullPointerException;

    void get(Consumer<String> session);

    void set(CreateSession session);

    boolean isPresent();
}

File: api/src/main/kotlin/hirez/SessionStorage.kt
package hirez

import hirez.json.CreateSession

interface SessionStorage {
	fun set(session: CreateSession)
	
	fun get(): String?
	
	fun remove()
	
	val isPresent: Boolean
	
	companion object {
		val DEFAULT = object : SessionStorage {
			private var key: String? = null
			
			override fun set(session: CreateSession) {
				this.key = session.sessionId
			}
			
			override fun get(): String? = key
			
			override fun remove() {
				key = null
			}
			
			override val isPresent = key != null
		}
	}
}
